File: twi_py/twi_post.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from auth_sga import CONSUMER_KEY,CONSUMER_SECRET,ACCESS_TOKEN,ACCESS_TOKEN_SECRET
#sga_auth には、上の4つの情報が入っています
import tweepy
import time

#oauth認証
auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
auth.set_access_token(ACCESS_TOKEN, ACCESS_TOKEN_SECRET)
api = tweepy.API(auth)

#post関数
def post(post_word):
    if post_word == "time":
        api.update_status(status = time.ctime())    
        return 1
    elif post_word == "exit":
        return -1
    else:
        api.update_status(status = post_word)
        return 1
def main():
    try:
        print("input post words")
        post_word = input()
        t = post(post_word)

        if t == 1:
            print("tweet success")
        elif t == -1:
            print("exit success")
        else:
            print("undefined error.")
        
    except KeyboardInterrupt:
        print("\nexit: KeyboardInterrupt")
        exit

if __name__ == "__main__":
    main()
    

File: twi_js/public/twi_post.js
var twitter = require('ntwitter');
var auth = require('../auth_sga.js');

var now = new Date();

//oauth認証
var tw = new twitter({
    consumer_key: auth.consumer_key(),
    consumer_secret: auth.consumer_secret(),
    access_token_key: auth.access_token_key(),
    access_token_secret: auth.access_token_secret()
});

//時間のついっと
function time_post(){
    
    var post_word = now.getHours() + "時" + now.getMinutes() + "分" + now.getSeconds() + "秒";
    
    //ついっと
    tw.updateStatus(post_word, function(er, data){console.log("tweet success: " + data.text);});
}
